File: rag-backend/src/main/java/cn/sdh/backend/entity/ScheduledTask.java
package cn.sdh.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("scheduled_task")
public class ScheduledTask implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String taskType;

    @TableField("cron_expression")
    private String cronExpression;

    private String params;

    private String description;

    private Byte status;

    @TableField("last_execute_time")
    private LocalDateTime lastExecuteTime;

    @TableField("next_execute_time")
    private LocalDateTime nextExecuteTime;

    @TableField("last_result")
    private String lastResult;

    @TableField("success_count")
    private Integer successCount;

    @TableField("fail_count")
    private Integer failCount;

    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("update_time")
    private LocalDateTime updateTime;
}

File: rag-frontend/src/api/scheduledTask.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types'

export interface ScheduledTask {
  id: number
  name: string
  taskType: string
  cronExpression: string
  params?: string
  description?: string
  status: number
  lastExecuteTime?: string
  nextExecuteTime?: string
  lastResult?: string
  successCount: number
  failCount: number
  createTime: string
}

export function getScheduledTasks(page: number = 1, pageSize: number = 10, taskType?: string, status?: number): Promise<ApiResponse<{ records: ScheduledTask[], total: number }>> {
  return request.get('/api/scheduled-task/list', { params: { page, pageSize, taskType, status } })
}

export function getScheduledTask(id: number): Promise<ApiResponse<ScheduledTask>> {
  return request.get(`/api/scheduled-task/${id}`)
}

export function createScheduledTask(data: Partial<ScheduledTask>): Promise<ApiResponse<ScheduledTask>> {
  return request.post('/api/scheduled-task', data)
}

export function updateScheduledTask(id: number, data: Partial<ScheduledTask>): Promise<ApiResponse<null>> {
  return request.post(`/api/scheduled-task/update/${id}`, data)
}

export function toggleTaskStatus(id: number): Promise<ApiResponse<null>> {
  return request.post(`/api/scheduled-task/toggle/${id}`)
}

export function executeTask(id: number): Promise<ApiResponse<null>> {
  return request.post(`/api/scheduled-task/${id}/execute`)
}

export function deleteScheduledTask(id: number): Promise<ApiResponse<null>> {
  return request.post(`/api/scheduled-task/delete/${id}`)
}
